frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -27,59 +27,46 @@ import { ProvisionSource } from '../../../../constants/vm/provision-source';
 import { prefillVmTemplateUpdater } from './prefill-vm-template-state-update';
 import { iGetPrameterValue, iGetAnnotation } from '../../../../selectors/immutable/common';
 import { CDI_UPLOAD_POD_ANNOTATION, CDI_UPLOAD_RUNNING } from '../../../cdi-upload-provider/consts';
+import { iGetLoadError } from '../../../../utils/immutable';
 
 const selectUserTemplateOnLoadedUpdater = (options: UpdateOptions) => {
   const { id, dispatch, getState } = options;
   const state = getState();
 
-  if (
-    iGetVmSettingAttribute(state, id, VMSettingsField.USER_TEMPLATE, 'initialized') ||
-    !options.changedCommonData.has(VMWizardProps.userTemplates)
-  ) {
-    return;
-  }
-
-  const userTemplateName = iGetCommonData(state, id, VMWizardProps.userTemplateName);
-  if (!userTemplateName) {
-    return;
+  const iUserTemplate = iGetLoadedCommonData(state, id, VMWizardProps.userTemplate);
+  if (iUserTemplate && iGetLoadError(iGetCommonData(state, id, VMWizardProps.userTemplate))) {
+    dispatch(
```

Comment:
This is for the use case when a template is deleted while you try to create a VM from that template - then we reset the initialization variable.

Reply:
I would not complicate it. Let's prefill it just once (that is also how vmimport works).

We are also not checking template updates at the moment.
Let's just assume the user can modify all the values once he clicks on the template.

The deletion of the template has to be notified though since we cannot continue with vm creation